### PR TITLE
Refactor OpenJPA and Eclipselink specific Introspection

### PR DESCRIPTION
**Pull request by jgrassel**
Signed-off-by: Joe Grassel <jgrassel@us.ibm.com>

Refactor OpenJPA introspection, add Eclipselink introspection.

### CODE REVIEW

**Comment by jgrassel** [comment]
#build

File: dev/com.ibm.ws.jpa.container.ormdiagnostics/src/com/ibm/ws/jpa/diagnostics/AbstractIntrospection.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    /*
+     * Utility methods used for introspecting JPA Persistence Provider Implementation Data Structures
+     */
+    protected static Object reflectObjValue(Object o, String field) throws Exception {
+        final Class<?> c = o.getClass();
+        final Field f = findField(c, field); //        c.getField(field);
+        if (f == null) {
+            return null;
+        }
+        final boolean accessible = f.isAccessible();
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.container.ormdiagnostics/src/com/ibm/ws/jpa/diagnostics/AbstractIntrospection.java]: Note: java.lang.Class.getDeclaredField will return a copy of the Field.... so changes made here with setAccessible are only visible here.... and do not need to be undone.

If it is possible to cache this Field, that would be nice, but probably not important if this won't run very often.

Will this run in a doPriv? or an a privileged thread? If running as regular trace, you may fail Java 2 security authorization.

Comment (jgrassel) [reply dev/com.ibm.ws.jpa.container.ormdiagnostics/src/com/ibm/ws/jpa/diagnostics/AbstractIntrospection.java]: I believe the introspection dump threads are privileged threads, I've been running the FATs (which enable java2sec) and the dump has not been failing with ACE involving reflection (only ACE is from the use of the jdk provided JAXB provider impl)

**Comment by jgrassel** [comment]
Failing fats not related to updated JPA diagnostic code.

**Pull request merged**